### PR TITLE
Connections: Add redirect notice for datasources and plugins pages

### PR DESCRIPTION
**Pull request by mikkancso**
**What is this feature?**

Add a redirect banner to the Plugins page when "All" or "Data sources" filter is set and to the Data sources page. The "Connections" button redirects to the corresponding page depending on the origin page.

[Screencast from 2023-01-05 13-18-45.webm](https://user-images.githubusercontent.com/13637610/210778919-898aca5b-833e-4ee6-9be5-8e55a1df23ec.webm)

**UPDATE:** after [this commit](https://github.com/grafana/grafana/pull/61037/commits/244399730ee5c6d0f61522c92b7b5acb4d50de71) the banner looks like this:
Screenshot from 2023-01-05 15-06-34

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/cloud-onboarding/issues/2886

### CODE REVIEW

**Review by anaivanov — APPROVED** [review]
LGTM

**Review by leventebalogh — APPROVED** [review]
Nice one! 🙌

File: public/app/features/connections/components/ConnectionsRedirectNotice/ConnectionsRedirectNotice.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Alert, LinkButton, useStyles2 } from '@grafana/ui';
+
+import { ROUTES } from '../../constants';
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  alert: css`
+  > div:nth-child(2) {
```
Comment (jarben) [inline public/app/features/connections/components/ConnectionsRedirectNotice/ConnectionsRedirectNotice.tsx:11]: Just wondering whether we want to rely on the internal implementation of the Alert component, what if we keep the text as title and set something like .alertButton {position: absolute; right: 10px} ?

Comment (mikkancso) [reply public/app/features/connections/components/ConnectionsRedirectNotice/ConnectionsRedirectNotice.tsx]: I fixed this, but then we're pretty much unable to make it look like it's designed. We can still get away with it, I think.

Comment (jarben) [reply public/app/features/connections/components/ConnectionsRedirectNotice/ConnectionsRedirectNotice.tsx]: I think you can add text as a title and position the button absolutely to make it look like before?

**Review by jarben — APPROVED** [review]
LGTM

File: public/app/features/connections/components/ConnectionsRedirectNotice/ConnectionsRedirectNotice.tsx (hunk 1 above)
Comment (mikkancso) [reply public/app/features/connections/components/ConnectionsRedirectNotice/ConnectionsRedirectNotice.tsx]: I'm not sure, but that also feels like a hack to me :/

**Pull request merged**